### PR TITLE
Add support for reversion, add tests and views for model forms

### PR DESCRIPTION
**Pull request by LegoStormtroopr**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -182,11 +185,11 @@ Django Garnett uses the python `langcodes` library to determine more information
 * `GARNETT_DEFAULT_TRANSLATABLE_LANGUAGE`
     * Stores the default language to be used for reading and writing fields if no language is set in a context manager or by a request.
     * By default it is 'en-AU' the [language code][term-language-code] for 'Strayan, the native tongue of inhabitants of 'Straya (or more commonly known as Australia). 
-    * Can also be callable that returns default language code
+    * This can also be callable that returns list of language codes. Combined with storing user settings in something like (django-solo)[https://github.com/lazybird/django-solo] users can dynamically add or change their language settings.
```
Comment (s-i-l-k-e) [inline README.md:188]: Be a callable

File: poetry.lock [hunk 2]
```diff
@@ -142,6 +150,29 @@ sqlparse = ">=0.2.2"
 argon2 = ["argon2-cffi (>=16.1.0)"]
 bcrypt = ["bcrypt"]
 
+[[package]]
+name = "django-reversion"
```
Comment (s-i-l-k-e) [inline poetry.lock:154]: Are these dev dependencies?

**Review by s-i-l-k-e — APPROVED** [review]
Looks good to me, just some small comments

File: poetry.lock (hunk 2 above)
Comment (LegoStormtroopr) [reply poetry.lock:154]: yeah, they are - I'll relock

File: garnett/reversion.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+def patch_compare():
+    from reversion_compare.compare import CompareObjects as COBase
+
+    class CompareObjects(COBase):
+        def __init__(self, field, field_name, obj, version1, version2, is_reversed):
+            if isinstance(field, TranslatedField):
+                field_name = field.attname
+            super().__init__(field, field_name, obj, version1, version2, is_reversed)
+
+    import reversion_compare.compare
```
Comment (s-i-l-k-e) [inline garnett/reversion.py:13]: Can we move this import to the top of the function?

File: garnett/serializers/base.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
+from django.core.serializers.base import Serializer as BaseSerializer
+from django.utils.encoding import is_protected_type
+from garnett.fields import TranslatedField
+
+
+def crazy(queryset):
```
Comment (s-i-l-k-e) [inline garnett/serializers/base.py:6]: This should probably have a doc-string explaining the craziness. It should probably also not be called 'crazy' even if it is crazy.

File: garnett/serializers/base.py [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+    def second_inner():
+        yield item
+        for t in a:
+            yield t
+
+    b = second_inner()
+    return item, b
+
+
+class TranslatableSerializer(BaseSerializer):
```
Comment (s-i-l-k-e) [inline garnett/serializers/base.py:20]: Do we still need all of this?

**Review by s-i-l-k-e — COMMENTED** [review]
LGTM, just a few small things to look at

File: garnett/reversion.py (hunk 3 above)
Comment (LegoStormtroopr) [reply garnett/reversion.py:13]: I think we want to import this as late as possible as we are monkey patching over it. I'll add a comment.

File: garnett/serializers/base.py (hunk 5 above)
Comment (LegoStormtroopr) [reply garnett/serializers/base.py:26]: no - its been removed.

File: garnett/serializers/base.py (hunk 4 above)
Comment (LegoStormtroopr) [reply garnett/serializers/base.py]: fixed.

**Pull request merged**